src/formatDate.js

```diff
@@ -50,7 +50,41 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const oldSeparator = fromFormat[3];
+  const partsDate = date.split(oldSeparator);
+
+  const objDate = {};
+
+  for (let i = 0; i < partsDate.length; i++) {
+    objDate[fromFormat[i]] = partsDate[i];
+  }
+
+  if (fromFormat.includes('YY') && toFormat.includes('YYYY')) {
+    if (objDate['YY'] < 30) {
+      objDate['YYYY'] = '20' + objDate['YY'];
+    } else {
+      objDate['YYYY'] = '19' + objDate['YY'];
+    }
+  }
+
+  if (fromFormat.includes('YYYY') && toFormat.includes('YY')) {
+    objDate['YY'] = objDate['YYYY'].slice(2);
+  }
+
+  const newDate = [];
+
+  for (const param of toFormat) {
+    newDate.push(objDate[param]);
+  }
+
+  // delete last element(undefined).
+  newDate.pop();
+
+  const newSeparator = toFormat[3];
```

Comment:
Consider using a more descriptive variable name like 'formattedDate' instead of 'updateDate'.